addons/vagrant/playbooks/linux_servers/samba4ad.yml

```diff
@@ -9,190 +9,217 @@
+
+    - name: Allow ntml auth in samba config
+      ansible.builtin.lineinfile:
+        path: /etc/hosts
+        regexp: '^{{ samba4ad__mgmt_ip }}\s+ad.example.lan\s+ad'
+        line: "{{ samba4ad__mgmt_ip }} ad.example.lan ad"
+ 
+    - name: Allow ntml auth in samba config
+      ansible.builtin.lineinfile:
+        path: /etc/samba/smb.conf
+        regexp: '^\s*ntlm auth = yes'
+        insertafter: '^\s*workgroup.*'
+        line: "        ntlm auth = yes"
+  
+    - name: Configure Samba admin password
+      ansible.builtin.expect:
+        command: "{{ item }}"
+        responses:
+          (?i)new\s(?i)password: "{{ samba4ad__password }}"
+          (?i)retype\s(?i)password: "{{ samba4ad__password }}"
+      loop:
+      - "samba-tool user setpassword {{ samba4ad__username }}"
+  
+    - name: Remove config files
+      file:
+        path: "{{ item }}"
+        state: absent
+      loop:
+      - "/var/lib/samba/private/krb5.conf"
+  
+    - name: Creating a symlink
+      ansible.builtin.file:
+        src: "/etc/krb5.conf"
+        dest: "/var/lib/samba/private/krb5.conf"
+        state: link
+  
+    - name: Unmask and start Services
+      ansible.builtin.systemd:
+        enabled: yes
+        state: restarted
+        masked: no
+        name: "{{ item }}"
+      loop:
+      - "samba-ad-dc" 
+  
+    - name: Reboot
+      ansible.builtin.reboot:
+  
+    - name: Wait for the reboot and reconnect
+      wait_for_connection:
+  
+    - name: Pause for 10 seconds after reboot
+      ansible.builtin.pause:
+        seconds: 10
+  
+    - name: Load kinit with admin password
+      ansible.builtin.expect:
+        command: "{{ item }}"
+        responses:
+          (?i)password.*: "{{ samba4ad__password }}"
+      loop:
+      - "kinit administrator"
+  
+    - name: Klist Samba
+      ansible.builtin.shell:
+        cmd: "{{ item }}"
+      loop:
+      - "klist"
+  
+    - name: Configure Samba "{{ packetfence_domain_account }}" user
+      ansible.builtin.expect:
+        command: "{{ item }}"
+        responses:
+          (?i)new\s(?i)password: "{{ packetfence_domain_password }}"
+          (?i)retype\s(?i)password: "{{ packetfence_domain_password }}"
+      loop:
+      - "samba-tool user add {{ packetfence_domain_account }} --username={{ packetfence_domain_account }} --mail-address={{ packetfence_domain_account }}@{{ dns_domain }} --workgroup={{ dns_domain_upper }}"
+
+    - name: Configure Samba AD Stuff
+      ansible.builtin.shell:
+        cmd: "{{ item }}"
+      loop:
+      - "samba-tool user setexpiry {{ samba4ad__username }} --noexpiry"
+      - "samba-tool user setexpiry {{ packetfence_domain_account }} --noexpiry"
+      - "samba-tool group add sponsor"
+      - "samba-tool group add IT"
+      - "samba-tool group addmembers sponsor {{ packetfence_domain_account }}"
+      - "samba-tool group addmembers IT {{ packetfence_domain_account }}"
+
+    when: check_if_samba.failed is true
```

Comment:
If `ansible.builtin.lineinfile` module returns "CHANGED" when it found a line in a file (even in check mode), you can certainly simplify this line by:

`when: check_if_samba is changed`

Which is, IMHO, more readable.

Reply:
This is not working when there is no installation.
To use that we need to change tasks order.